src/test/java/com/cag/cagbackendapi/integration/ProfileControllerIntegrationTests.kt

```diff
@@ -186,4 +195,121 @@ class ProfileControllerIntegrationTests {
+        headers2.set("authKey", validAuthKey)
+        headers2.set("userId", userIdUUID.toString())
+        val request2 = HttpEntity(userProfile, headers2)
+
+        //create profile
+        val userProfileResponse = testRestTemplate.exchange("/user/${userIdUUID.toString()}/profile/register", HttpMethod.POST, request2, String::class.java)
+        val createdProfile = objectMapper.readValue(userProfileResponse.body, ProfileDto::class.java)
+
+        //get profile headers
+        val headers3 = HttpHeaders()
+        headers3.set("authKey", validAuthKey)
+        headers3.set("userId", userIdUUID.toString())
+        val request3 = HttpEntity(null, headers3)
+
+        //get profile
+        val getProfileResponse = testRestTemplate.exchange("/user/${userIdUUID.toString()}/profile", HttpMethod.GET, request3, String::class.java)
+        val getProfile = objectMapper.readValue(getProfileResponse.body, ProfileDto::class.java)
+
+        //check the created user
+        assertNotNull(createdUserResponse)
+        assertEquals(HttpStatus.CREATED, createdUserResponse.statusCode)
+        Assertions.assertEquals(validRegisterUser.first_name, createUser.first_name)
+        Assertions.assertEquals(validRegisterUser.last_name, createUser.last_name)
+        Assertions.assertEquals(validRegisterUser.email, createUser.email)
+        assertNotNull(createUser.userId)
+
+        //get the created profile
+        assertNotNull(getProfileResponse)
+        assertEquals(HttpStatus.OK, getProfileResponse.statusCode)
+        assertNotNull(getProfile.profile_id)
+        assertEquals(userProfile.pronouns, getProfile.pronouns)
+        assertEquals(userProfile.lgbtqplus_member, getProfile.lgbtqplus_member)
+        assertEquals(userProfile.gender_identity, getProfile.gender_identity)
+        assertEquals(userProfile.comfortable_playing_man, getProfile.comfortable_playing_man)
+        assertEquals(userProfile.comfortable_playing_women, getProfile.comfortable_playing_women)
+        assertEquals(userProfile.comfortable_playing_neither, getProfile.comfortable_playing_neither)
+        assertEquals(userProfile.comfortable_playing_transition, getProfile.comfortable_playing_transition)
+        assertEquals(userProfile.height_inches, getProfile.height_inches)
+        assertEquals(userProfile.agency, getProfile.agency)
+        assertEquals(userProfile.website_link_one, getProfile.website_link_one)
+        assertEquals(userProfile.website_link_two, getProfile.website_link_two)
+        assertEquals(userProfile.website_type_one, getProfile.website_type_one)
+        assertEquals(userProfile.website_type_two, getProfile.website_type_two)
+        assertEquals(userProfile.bio, getProfile.bio)
+    }
+
+    @Test
+    fun getProfile_invalidUserId_400BadRequest() {
+        val userId = null
+
+        //get profile headers
+        val headers3 = HttpHeaders()
+        headers3.set("authKey", validAuthKey)
+        headers3.set("userId", userId)
```

Comment:
The UserId is sent in the path, not in the header! You also have it in the path so this test will pass, but this header is just extra unneeded data